spacemeshos/smapp
# Fix key pair derivation: do it the same as smcli does

It fixes #1237 and closes #1136
Now Smapp derives key pairs in the same way as smcli in this PR: https://github.com/spacemeshos/smcli/pull/37
Also, it removes the dependency on a custom `ed25519.wasm`.

### How to test
1. Clone `smcli/derivation`
2. Build smcli (`make build`)
3. Run Smapp, create the wallet and copy-paste mnemonics
4. Read wallet file by smcli: `smcli wallet read -f ~/path/to/wallet.json`
5. Then create wallet using same mnemonics with smcli: `smcli wallet create`, paste mnemonics, and type password
6. Read wallet file: `smcli wallet read -f ~/.spacemesh/newly-created-wallet.json`
7. Public key of `Main account` of Smapp-created wallet should be equal to `Child Key 0`

### Important notes
There are a couple of moments needed which need to be checked:

1. The secret key of the derived key pair is now 64 bytes in length, whereas the second 32 bytes is the public key.
   Not sure it will work correctly for signing/verifying on the go-spacemesh side, can't check it now, has zero balance on both networks.
2. By default `smcli` generates mnemonics using 24 words, while Smapp uses only 12 words. As a consequence now Smapp can't import the wallet created by smcli (but it works in another direction).
   Seems this is a candidate for the next improvement, so I just want to know what is our plan: (a) always use 24 words, (b) support both.
